<!-- -*- mode: xml -*- -->
<launch>
  <arg name="INPUT" default="/rgb_input"/>
  <arg name="OUTPUT" default="/rgb_output"/>
  <arg name="DEFAULT_NAMESPACE" default="RGB_color_filter"/>

  <arg name="r_max" default="255" />
  <arg name="r_min" default="0" />
  <arg name="g_max" default="255" />
  <arg name="g_min" default="0" />
  <arg name="b_max" default="255" />
  <arg name="b_min" default="0" />

  <arg name="create_manager" default="true" />
  <arg name="manager" default="rgb_filter_manager" />

  <group ns="$(arg DEFAULT_NAMESPACE)">
    <node if="$(arg create_manager)"
          pkg="nodelet" type="nodelet" name="$(arg manager)"
          args="manager" output="screen"/>

    <node pkg="nodelet" type="nodelet" name="rgb_filter"
          args="load jsk_pcl/RGBColorFilter $(arg manager)" output="screen">
      <remap from="~input" to="$(arg INPUT)" />
      <remap from="~output" to="$(arg OUTPUT)" />
      <rosparam>
        use_indices: false
      </rosparam>
      <param name="r_limit_max" value="$(arg r_max)" />
      <param name="r_limit_min" value="$(arg r_min)" />
      <param name="g_limit_max" value="$(arg g_max)" />
      <param name="g_limit_min" value="$(arg g_min)" />
      <param name="b_limit_max" value="$(arg b_max)" />
      <param name="b_limit_min" value="$(arg b_min)" />
    </node>

  </group>

</launch>
